lib/config/configitem.cpp

```diff
@@ -450,180 +450,139 @@ bool ConfigItem::CommitNewItems(const ActivationContext::Ptr& context, WorkQueue
 		<< "Committing " << total << " new items.";
 #endif /* I2_DEBUG */
 
-	std::set<Type::Ptr> types;
-	std::set<Type::Ptr> completed_types;
+	std::vector<Type::Ptr> types;
 	int itemsCount {0};
 
 	for (const Type::Ptr& type : Type::GetAllTypes()) {
 		if (ConfigObject::TypeInstance->IsAssignableFrom(type))
-			types.insert(type);
+			types.emplace_back(type);
```

Comment:
Yes, we should address such in

* #8753

one nice day.

Reply:
I wasn't suggesting you to remove this check, I was just pointing out that you are unnecessarily copying around the types and should avoid doing that. It's not something that already existed and simply saying it will be fixed "one nice day" isn't an argument.

Reply:
But not all types are config types.

Reply:
CONFIG items get committed here, of course only CONFIG types are relevant here. Btw. that preserved check IS "something that already existed" (just like https://github.com/Icinga/icinga2/pull/10003#discussion_r1491148736).

Reply:
I don't think you get my point her! My suggestion is to drop [`this loop`](https://github.com/Icinga/icinga2/blob/972662081d2698961acfb833bf6ae2755d91f1b9/lib/config/configitem.cpp#L456-L459) entirely and move the check [down here](https://github.com/Icinga/icinga2/blob/972662081d2698961acfb833bf6ae2755d91f1b9/lib/config/configitem.cpp#L463-L465) instead. Your sort method shouldn't care whether a type is assignable to a config object or not.